src/linear_algebra/projection.lean

```diff
@@ -167,6 +167,101 @@ lemma linear_proj_of_is_compl_idempotent (h : is_compl p q) (x : E) :
     linear_proj_of_is_compl p q h x :=
 linear_proj_of_is_compl_apply_left h _
 
+-- ↓ Remove after Eric's PR
+lemma equiv.exists_unique {α β : Type*} (e : α ≃ β) (b : β) : ∃! (a : α), e a = b :=
+⟨e.symm b, e.apply_symm_apply _, λ a h, (equiv.eq_symm_apply e).mpr h⟩
+
+lemma exists_unique_add_of_is_compl_prod (hc : is_compl p q) (x : E) :
+  ∃! (u : p × q), (u.fst : E) + u.snd = x :=
+equiv.exists_unique (prod_equiv_of_is_compl _ _ hc).to_equiv _
+
+lemma exists_unique_add_of_is_compl (hc : is_compl p q) (x : E) :
+  ∃ (u : p) (v : q), ((u : E) + v = x ∧ ∀ (r : p) (s : q),
+    (r : E) + s = x → r = u ∧ s = v) :=
+let ⟨u, hu₁, hu₂⟩ := exists_unique_add_of_is_compl_prod hc x in
+  ⟨u.1, u.2, hu₁, λ r s hrs, prod.eq_iff_fst_eq_snd_eq.1 (hu₂ ⟨r, s⟩ hrs)⟩
+
+/-- Given linear maps `φ` and `ψ` from complement submodules, `of_is_compl` is
+the induced linear map over the entire module. -/
+def of_is_compl (h : is_compl p q) (φ : p →ₗ[R] F) (ψ : q →ₗ[R] F) : E →ₗ[R] F :=
+φ.comp (linear_proj_of_is_compl p q h) + ψ.comp (linear_proj_of_is_compl q p h.symm)
```

Comment:
I'm curious if the proofs below become easier with:
```
(linear_map.coprod φ ψ).comp (submodule.prod_equiv_of_is_compl _ _ h).symm
```
Note that `linear_proj_of_is_compl` is defined in terms of `prod_equiv_of_is_compl` anyway

Reply:
I will give that a try!

Reply:
It definitely seems that your definition is easier to work with and I was able to shorten the proofs quite a bit :)